src/rules/groupedImportsRule.ts

```diff
@@ -0,0 +1,152 @@
+/**
+ * @license
+ * Copyright 2017 Palantir Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as Lint from "tslint";
+import { isImportDeclaration } from "tsutils";
+import * as ts from "typescript";
+
+export class Rule extends Lint.Rules.AbstractRule {
+    /* tslint:disable:object-literal-sort-keys */
+    public static metadata: Lint.IRuleMetadata = {
+        ruleName: "grouped-imports",
+        description: "Separate import groups by blank lines.",
+        rationale: "Keeps a clear overview on dependencies.",
+        optionsDescription: "Not configurable.",
+        hasFix: true,
+        options: {},
+        optionExamples: [true],
+        type: "style",
+        typescriptOnly: false,
+    };
+    /* tslint:enable:object-literal-sort-keys */
+
+    public static IMPORT_SOURCES_SEPARATED = "Import sources within a group must not be separated by blank lines";
+    public static IMPORT_SOURCES_NOT_SEPARATED =
+        "Import sources of different groups must be separated by a single blank line";
+    public static IMPORT_SOURCES_ORDER =
+        "Import sources of different groups must be sorted by: libraries, parent directories, current directory";
+
+    public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+        return this.applyWithWalker(new Walker(sourceFile, this.ruleName, this.getOptions()));
+    }
+}
+
+enum ImportStatementType {
+    LIBRARY_IMPORT = 1,
+    PARENT_DIRECTORY_IMPORT = 2,
+    CURRENT_DIRECTORY_IMPORT = 3,
+}
+
+interface ImportStatement {
+    statement: ts.Statement;
+    type: ImportStatementType;
+    lineStart: number;
+    lineEnd: number;
+}
+
+class Walker extends Lint.AbstractWalker<Lint.IOptions> {
+    private lastImportStatement: ImportStatement;
```

Comment:
Is this really necessary? It adds a lot of complexity and additional testing throughout the rest of the code.